mk/leaf.mk

```diff
@@ -16,7 +16,7 @@ export GO_COMPILE_FLAGS=-gcflags '-N -l'
 endif
 
 $(EXECUTABLES) : % : %.go
-	go build $(GO_COMPILE_FLAGS) $< $(COMMON_FILES)
+	go build -tags hyperscan_v4 $(GO_COMPILE_FLAGS) $< $(COMMON_FILES)
```

Comment:
I don't think it is a good approach to add `-tags hyperscan_v4` to a general build rule used to build all Go sources in the framework. Instead I suggest that you define `GO_COMPILE_FLAGS=-tags hyperscan_v4` in the beginning of `examples/dpi/pattern/Makefile` before including `leaf.mk`. This should add necessary switch only to compilation of DPI example.

Reply:
really good catch!